pkgs/development/coq-modules/vscoq-language-server/default.nix

```diff
@@ -0,0 +1,32 @@
+{ metaFetch, mkCoqDerivation, coq, lib, glib, gnome, wrapGAppsHook,
+  version ? null }:
+
+let ocamlPackages = coq.ocamlPackages;
+    defaultVersion = lib.switch coq.coq-version [
+      { case = "8.18"; out = "2.0.3+coq8.18"; }
+    ] null;
+    location = { domain = "github.com"; owner = "coq-community"; repo = "vscoq"; };
+    fetch = metaFetch ({
+      release."2.0.3+coq8.18".sha256 = "sha256-VXhHCP6Ni5/OcsgoI1EbJfYCpXzwkuR8kbbKrl6dfjU=";
+      release."2.0.3+coq8.18".rev = "v2.0.3+coq8.18";
+      inherit location; });
+    fetched = fetch (if version != null then version else defaultVersion);
+in
+ocamlPackages.buildDunePackage {
+  pname = "vscoq-language-server";
+  inherit (fetched) version;
+  src = "${fetched.src}/language-server";
+  buildInputs =
+    [ coq glib gnome.adwaita-icon-theme wrapGAppsHook ] ++
+    (with coq.ocamlPackages; [ findlib
```

Comment:
```suggestion
    (with ocamlPackages; [
```

Reply:
On my machine if I don't specify coq.ocamlPackages vscoq-language-server does not build, it no longer finds coq-core.

Reply:
Plus if we do not add findlib it does not work either.

Reply:
Nevermind ... I rebound it earlier in the file already

Reply:
@vbgl indeed I can remove `coq.` (since I remapped `ocamlPackages` in the begining)
However, `findlib` being (by default as you expected) in `nativeBuildInputs` does not suffice, and I do not understand why :thinking: 
```
File "/nix/store/wx660wwhvgmrxrc88c6g2p31x3wixxj2-coq-8.18.0/lib/ocaml/4.14.1/site-lib/coq-core/dune-package", line 6614, characters 41-48:
6614 |  (requires coq-core.boot coq-core.vernac findlib)
                                                ^^^^^^^
Error: Library "findlib" not found.
-> required by library "coq-core.sysinit" in
   /nix/store/wx660wwhvgmrxrc88c6g2p31x3wixxj2-coq-8.18.0/lib/ocaml/4.14.1/site-lib/coq-core/sysinit
-> required by executable vscoqtop in vscoqtop/dune:2
-> required by _build/default/vscoqtop/vscoqtop.exe
-> required by _build/install/default/bin/vscoqtop
-> required by _build/default/vscoq-language-server.install
-> required by alias install
```

Reply:
Ok. Fine then. Sorry for the wrong suggestion.

Reply:
Your suggestion actually looks fine... I would prefer to understand why it does not work.

Reply:
When `findlib` is in `nativeBuildInputs`, the `ocamlfind` tool runs & works fine on the host machine, but you cannot link against the `findlib.cmx` etc. libraries that are meant for the target machine. If you want to link against some libraries, you need them as `buildInputs` rather than `nativeBuildInputs`.